### PR TITLE
Replicate number format change

### PR DESCRIPTION
**Pull request by seenanair**
Replicate number must be a string of up to 7 letters and numbers.

### CODE REVIEW

File: cypress/e2e/shared/registration.cy.ts [hunk 1]
```diff
@@ -68,30 +68,18 @@ export function shouldBehaveLikeARegistrationForm(registrationType: Registration
...
-      cy.findByTestId('Replicate Number').type('0').blur();
-      checkReplicateWarningIsVisible();
-    });
-    it('requires Replicate Number to be number or number followed by a lowercase letter', () => {
-      cy.findByTestId('Replicate Number').type('0ab').blur();
-      checkReplicateWarningIsVisible();
-    });
-    it('requires Replicate Number to be number or number followed by a lowercase letter', () => {
-      cy.findByTestId('Replicate Number').type('5ab').blur();
+      cy.findByTestId('Replicate Number').clear().type('1ab6789').blur();
```
Comment (khelwood) [inline cypress/e2e/shared/registration.cy.ts:78]: Why should a warning be shown for the replicate value "1ab6789"?

File: src/lib/validation/registrationValidation.ts [hunk 2]
```diff
@@ -111,14 +111,14 @@ export default class RegistrationValidation {
     if (this.tissueSampleRegistration) {
       return validation.optionalString({
         label: 'Replicate Number',
-        restrictChars: /^[1-9]\d*[a-z]?$/,
-        errorMessage: 'Replicate Number must be a positive integer, optionally followed by a lower case letter.'
+        restrictChars: /^([a-zA-Z0-9]){1,7}$/,
```
Comment (khelwood) [inline src/lib/validation/registrationValidation.ts:114]: Should probably be `/^[a-zA-Z0-9]{1,7}$/`

File: src/lib/validation/registrationValidation.ts [hunk 3]
```diff
@@ -111,14 +111,14 @@ export default class RegistrationValidation {
...
+        errorMessage: 'Replicate number must be a string of up to 7 letters and numbers.'
       });
     } else {
       return validation.requiredString({
         label: 'Replicate Number',
-        restrictChars: /^[1-9]\d*[a-z]?$/,
-        errorMessage: 'Replicate Number must be a positive integer, optionally followed by a lower case letter.'
+        restrictChars: /^([a-zA-Z0-9]){1,7}$/,
```
Comment (khelwood) [inline src/lib/validation/registrationValidation.ts:120]: Should probably be `/^[a-zA-Z0-9]{1,7}$/`

File: cypress/e2e/shared/registration.cy.ts (hunk 1 above)
Comment (seenanair) [reply cypress/e2e/shared/registration.cy.ts:78]: That's a mistake, thought it was >7. Chnaged it now. But, I wonder why it worked. Anyway, rechecked it to make sure it validates the correct rule and it all seems to work

**Review by khelwood — APPROVED** [review]

**Pull request merged**